language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -702,6 +703,46 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
 //        return super.transform(interpolationNode);
 //    }
 
+    @Override
+    public Optional<TypeSymbol> transform(FromClauseNode fromClauseNode) {
+        if (context.currentSemanticModel().isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (context.getCursorPositionInTree() > fromClauseNode.inKeyword().textRange().endOffset()) {
+            Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get().typeOf(fromClauseNode.expression());
+
+            if (typeSymbol.isPresent()) {
+                return typeSymbol;
+            }
+
+            Optional<Symbol> optionalSymbol = context.currentSemanticModel().get()
+                    .symbol(fromClauseNode.typedBindingPattern().bindingPattern());
+            if (optionalSymbol.isEmpty()) {
+                return Optional.empty();
+            }
+
+            typeSymbol = SymbolUtil.getTypeDescriptor(optionalSymbol.get());
+            if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+                return Optional.empty();
+            }
+            return Optional.of(context.currentSemanticModel()
+                    .get().types().builder().ARRAY_TYPE.withType(typeSymbol.get()).build());
+        }
+
+        Optional<Symbol> optionalSymbol = context.currentSemanticModel().get()
+                .symbol(fromClauseNode.typedBindingPattern().bindingPattern());
+        if (optionalSymbol.isEmpty()) {
+            return context.currentSemanticModel().get()
+                    .typeOf(fromClauseNode.expression())
+                    .filter(typeSymbol -> typeSymbol.typeKind() == TypeDescKind.ARRAY)
```

Comment:
It can be one of the following. So filtering just for array is not accurate.

TypeDescKind.STRING,
TypeDescKind.ARRAY,
TypeDescKind.MAP,
TypeDescKind.TABLE,
TypeDescKind.STREAM,
TypeDescKind.XML